libs/blocks/global-navigation/global-navigation.js

```
@@ -74,17 +74,27 @@ export const CONFIG = {
         name: 'profile',
         attributes: {
           isSignUpRequired: false,
+          messageEventListener: (event) => {
+            const { name, payload, executeDefaultAction } = event.detail;
+            if (name === 'System' && payload.subType === 'AppInitiated') {
```

We can fail fast here
```js
if (name !== 'System') return;
switch (payload.subType) {
    case 'AppInitiated':
        // code
    case 'SignOut':
        // code
    default:
        // code
}

```

---

Thank you @narcis-radu . Done. Please review.